File: CS/T191208/CustomControl.cs
using System.Drawing;
using System.Windows.Forms;
using System;

using DevExpress.XtraBars.Customization;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraBars;
using DevExpress.XtraTreeList;

namespace T191208 {
    public partial class CustomCustomizationControl : CustomizationControl, ISupportCustomizationDrag {
        // Fields...
        Control selectedControl;
        protected Control SelectedControl {
            get {
                return selectedControl;
            }
            set {
                selectedControl = value;
            }
        }

        public CustomCustomizationControl() {
            InitializeComponent();
            this.gridControl1.MouseDown += gridControl1_MouseDown;
            this.gridControl2.MouseDown += gridControl2_MouseDown;
            this.treeList1.MouseDown += treeList1_MouseDown;
        }

        private void CustomCustomizationControl_Load(object sender, EventArgs e) {

        }

        void treeList1_MouseDown(object sender, MouseEventArgs e) {
            SelectedControl = sender as TreeList;
        }

        void gridControl1_MouseDown(object sender, MouseEventArgs e) {
            SelectedControl = sender as GridControl;
        }

        private void gridControl2_MouseDown(object sender, MouseEventArgs e) {
            SelectedControl = sender as GridControl;
        }
        public void AddMouseMoveHandler(MouseEventHandler mouseMove) {
            this.gridView2.MouseMove += mouseMove;
            this.gridView1.MouseMove += mouseMove;
            this.treeList1.MouseMove += mouseMove;
        }


        public void AddMouseDownHandler(MouseEventHandler mouseDown) {
            this.gridView2.MouseDown += mouseDown;
            this.gridView1.MouseDown += mouseDown;
            this.treeList1.MouseDown += mouseDown;
        }

        public void AddMouseUpHandler(MouseEventHandler mouseUp) {
            this.gridView2.MouseUp += mouseUp;
            this.gridView1.MouseUp += mouseUp;
            this.treeList1.MouseUp += mouseUp;
        }

        public BarItem GetDraggingItem(Point location) {
            // your logic here
            BarItem item = null;
            if (SelectedControl is GridControl) {
                GridControl grid = SelectedControl as GridControl;
                if (grid != null) {
                    GridView view = grid.MainView as GridView;
                    if (view != null) {
                        GridHitInfo hitInfo = view.CalcHitInfo(location);
                        item = view.GetRow(hitInfo.RowHandle) as BarItem;
                    }
                }
            }
            if (SelectedControl is TreeList) {
                TreeList treeList = SelectedControl as TreeList;
                if (treeList != null) {
                    TreeListHitInfo hitInfo = treeList.CalcHitInfo(location);
                    item = treeList1.GetDataRecordByNode(hitInfo.Node) as BarItem;
                }
            }
            return item;
        }

        public Control GetControlToPerformDragging(Point location) {
            //your logic here
            return SelectedControl;
        }
        public void InitControl(BarManager manager) {
            this.gridControl1.DataSource = manager.Items;
            this.gridControl2.DataSource = manager.Items;
            this.treeList1.DataSource = manager.Items;
        }
    }
}

File: VB/T191208/CustomControl.vb
Imports System.Drawing
Imports System.Windows.Forms
Imports System

Imports DevExpress.XtraBars.Customization
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraBars
Imports DevExpress.XtraTreeList

Namespace T191208
    Partial Public Class CustomCustomizationControl
        Inherits CustomizationControl
        Implements ISupportCustomizationDrag

        ' Fields...

        Private selectedControl_Renamed As Control
        Protected Property SelectedControl() As Control
            Get
                Return selectedControl_Renamed
            End Get
            Set(ByVal value As Control)
                selectedControl_Renamed = value
            End Set
        End Property

        Public Sub New()
            InitializeComponent()
            AddHandler Me.gridControl1.MouseDown, AddressOf gridControl1_MouseDown
            AddHandler Me.gridControl2.MouseDown, AddressOf gridControl2_MouseDown
            AddHandler Me.treeList1.MouseDown, AddressOf treeList1_MouseDown
        End Sub

        Private Sub CustomCustomizationControl_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

        End Sub

        Private Sub treeList1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            SelectedControl = TryCast(sender, TreeList)
        End Sub

        Private Sub gridControl1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            SelectedControl = TryCast(sender, GridControl)
        End Sub

        Private Sub gridControl2_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            SelectedControl = TryCast(sender, GridControl)
        End Sub
        Public Sub AddMouseMoveHandler(ByVal mouseMove As MouseEventHandler) Implements ISupportCustomizationDrag.AddMouseMoveHandler
            AddHandler Me.gridView2.MouseMove, mouseMove
            AddHandler Me.gridView1.MouseMove, mouseMove
            AddHandler Me.treeList1.MouseMove, mouseMove
        End Sub


        Public Sub AddMouseDownHandler(ByVal mouseDown As MouseEventHandler) Implements ISupportCustomizationDrag.AddMouseDownHandler
            AddHandler Me.gridView2.MouseDown, mouseDown
            AddHandler Me.gridView1.MouseDown, mouseDown
            AddHandler Me.treeList1.MouseDown, mouseDown
        End Sub

        Public Sub AddMouseUpHandler(ByVal mouseUp As MouseEventHandler) Implements ISupportCustomizationDrag.AddMouseUpHandler
            AddHandler Me.gridView2.MouseUp, mouseUp
            AddHandler Me.gridView1.MouseUp, mouseUp
            AddHandler Me.treeList1.MouseUp, mouseUp
        End Sub

        Public Function GetDraggingItem(ByVal location As Point) As BarItem Implements ISupportCustomizationDrag.GetDraggingItem
            ' your logic here
            Dim item As BarItem = Nothing
            If TypeOf SelectedControl Is GridControl Then
                Dim grid As GridControl = TryCast(SelectedControl, GridControl)
                If grid IsNot Nothing Then
                    Dim view As GridView = TryCast(grid.MainView, GridView)
                    If view IsNot Nothing Then
                        Dim hitInfo As GridHitInfo = view.CalcHitInfo(location)
                        item = TryCast(view.GetRow(hitInfo.RowHandle), BarItem)
                    End If
                End If
            End If
            If TypeOf SelectedControl Is TreeList Then
                Dim treeList As TreeList = TryCast(SelectedControl, TreeList)
                If treeList IsNot Nothing Then
                    Dim hitInfo As TreeListHitInfo = treeList.CalcHitInfo(location)
                    item = TryCast(treeList1.GetDataRecordByNode(hitInfo.Node), BarItem)
                End If
            End If
            Return item
        End Function

        Public Function GetControlToPerformDragging(ByVal location As Point) As Control Implements ISupportCustomizationDrag.GetControlToPerformDragging
            'your logic here
            Return SelectedControl
        End Function
        Public Sub InitControl(ByVal manager As BarManager) Implements ISupportCustomizationDrag.InitControl
            Me.gridControl1.DataSource = manager.Items
            Me.gridControl2.DataSource = manager.Items
            Me.treeList1.DataSource = manager.Items
        End Sub
    End Class
End Namespace
